src/map/lua/lua_baseentity.cpp

```diff
@@ -366,6 +366,33 @@ inline int32 CLuaBaseEntity::isJugPet(lua_State* L)
 
 //======================================================//
 
+inline int32 CLuaBaseEntity::getCharmChance(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+    
+    auto PCharmer = static_cast<CBattleEntity*>(m_PBaseEntity);
+    auto PTarget = static_cast<CBattleEntity*>(PLuaBaseEntity->GetBaseEntity());
```

Comment:
I originally had dynamic_cast and the null check in CLuaBaseEntity::getCharmChance because I figured someone could possibly pass in an npc/baseentity from lua (even if accidental).